packages/mds-api-server/index.ts

```diff
@@ -9,7 +9,18 @@ import { Params, ParamsDictionary } from 'express-serve-static-core'
 
 export type ApiRequest<P extends Params = ParamsDictionary> = express.Request<P>
 
-export type ApiQuery<Q extends string> = { query: Partial<{ [P in Q]: string | string[] }> }
+export type ApiQuery<Q1 extends string, Q2 extends string[] = never> = {
+  query: Partial<
+    {
+      [P in Q1]: string
+    }
+  > &
+    Partial<
+      {
+        [P in Q2[number]]: string | string[]
+      }
+    > & { [x: string]: never }
```

Comment:
What's the purpose of this `never` type in `ApiQuery`?

Reply:
It forces you to specify all of the expected query string parameters. It's best to illustrate with an example:
```ts
const req: ApiRequest & ApiQuery<'one' | 'only', ['many', 'more']> = ...
const { one, only, many, more, idk } = req.query
```
Gets typed as 
```ts
{
   one: string | undefined
   only: string | undefined
   many: string | string[] | undefined
   more: string | string[] | undefined
   idk: never
}
```
So, basically, any property other than those in the type declaration gets typed as `never` meaning you really can't do anything with it.